core/src/main/java/hudson/model/Queue.java

```diff
@@ -1535,32 +1575,27 @@ public void maintain() {
             ConsistentHash<Node> hash = new ConsistentHash<Node>(NODE_HASH);
             hash.addAll(hashSource);
 
-            Label lbl = p.getAssignedLabel();
             for (Node n : hash.list(p.task.getFullDisplayName())) {
                 final Computer c = n.toComputer();
-                if (c==null || c.isOffline())    continue;
-                if (lbl!=null && !lbl.contains(n))  continue;
-                if (n.canTake(p) != null) continue;
+                if (c == null || c.isOffline()) {
+                    continue;
+                }
+                if (n.canTake(p) != null) {
```

Comment:
It's rather a :bug: for me, because <code>Node:canTake()</code> is overridable => An abstract plugin developer may implement Node improperly and cause the scheduling incorrectness on the entire Jenkins instance